exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_client_options.h

```diff
@@ -43,6 +45,9 @@ struct OtlpGrpcClientOptions
   std::string ssl_client_cert_string;
 #endif
 
+  /** Use custom ChannelCredentials, instead of the SSL options above. */
+  std::shared_ptr<grpc::ChannelCredentials> credentials;
```

Comment:
std::optional instead? (or the `nostd` one?)

Reply:
I don’t think we need the complexity of wrapping it in an optional. It would lead to two different unset states – `std::optional` set to `nullopt`, and `std::optional` containing a `nullptr`.

`std::shared_ptr` defaults to `nullptr` to represent the unset state, similar to how other unset options are represented as empty strings.

This is passed to `grpc::CreateCustomChannel` that takes a `std::shared_ptr<grpc::ChannelCredentials>`.

Reply:
`std::optional` is not available in C++14, which is minimum requirement for the SDK.